### ISSUE TITLE
SqlDateTime overflow when selecting null column

### ISSUE THREAD

**Issue by bleroy** [bleroy/OrchardImportTest#2258]
I have a table with nullable DateTime column. When I use IRepository.Get I get
a SqlDateTime overflow exception whenever I select a record with null value in
the DateTime field.  
  
SqlDateTime overflow. Must be between 1/1/1753 12:00:00 AM and 12/31/9999
11:59:59 PM.  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 18368](http://orchard.codeplex.com/workitem/18368)
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Jan 18, 2012 at 3:11 PM
**Reported by:**	Emil
**Updated on:**	Jan 19, 2012 at 11:49 PM
**Updated by:**	BertrandLeRoy
**Closed on:**	Jan 19, 2012 at 11:49 PM
**Closed by:**	BertrandLeRoy
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Jan 18, 2012 at 3:11 PM
**Reported by:**	Emil
**Updated on:**	Jan 19, 2012 at 11:49 PM
**Updated by:**	BertrandLeRoy
**Closed on:**	Jan 19, 2012 at 11:49 PM
**Closed by:**	BertrandLeRoy

**Comment by bleroy** [comment]
On *2012-01-19 00:37:17 UTC*, **@sebastienros** commented:

Did you check that the model also has a DateTime? property ? Otherwise it will fail as DateTime is a value type.

**Comment by bleroy** [comment]
On *2012-01-19 03:43:16 UTC*, **@BertrandLeRoy** commented:

Sébastien means a nullable DataTime, the question mark is really important here.

**Comment by bleroy** [comment]
On *2012-01-19 23:31:47 UTC*, **@Emil** commented:

Sorry, my bad. I was misled from the exception because I got it on SELECT and didn't realized the case. Usually such exception might happen when you try to insert/update the value, but it is actually possible in the other way around too.

Once I changed the type to DateTime? it works fine.
Consider this WI closed and thanks for your invaluable help again.
